__tests__/src/test/java/org/example/RedemptionsTest.java

```diff
@@ -0,0 +1,169 @@
+import io.voucherify.client.model.*;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.io.IOException;
+import java.math.BigDecimal;
+
+import org.json.JSONException;
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.example.helpers.JsonHelper;
+
+@org.junit.jupiter.api.Order(8)
+public class RedemptionsTest {
+    public static ApiClient defaultClient = null;
+    public static RedemptionsApi redemptions = null;
+    public static CampaignsApi campaigns = null;
+
+    @BeforeAll
+    public static void beforeAll() {
+        defaultClient = Utils.getClient();
+        redemptions = new RedemptionsApi(defaultClient);
+        campaigns = new CampaignsApi(defaultClient);
+    }
+
+    @Test
+    @org.junit.jupiter.api.Order(1)
+    public void listRedemptionsTest() {
+        RedemptionsApi redemptions = new RedemptionsApi(defaultClient);
+
+        try {
+            RedemptionsListResponseBody responseBody = redemptions.listRedemptions(
+                    1,
+                    2,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null);
+
+            assertNotNull(responseBody);
+        } catch (ApiException | JsonSyntaxException e) {
+            fail();
+        }
+    }
+
+    @Test
+    @org.junit.jupiter.api.Order(2)
+    public void redeemStackedApplicableDiscountsTest() {
+        String snapshotPath = "src/test/java/org/example/snapshots/Redemptions/ApplicableRedemption.snapshot.json";
+        RedemptionsRedeemRequestBody redemptionsBody = getRedemptionsRequestBody(1);
+        redeemStackedDiscounts(redemptionsBody, snapshotPath);
+    }
+
+    @Test
+    @org.junit.jupiter.api.Order(3)
+    public void redeemStackedSkippedDiscountsTest() {
+        String snapshotPath = "src/test/java/org/example/snapshots/Redemptions/SkippedRedemption.snapshot.json";
+        RedemptionsRedeemRequestBody redemptionsBody = getRedemptionsRequestBody(30);
+        redeemStackedDiscounts(redemptionsBody, snapshotPath);
+    }
+
+    private void redeemStackedDiscounts(RedemptionsRedeemRequestBody requestBody, String snapshotPath) {
+        try {
+            RedemptionsRedeemResponseBody responseBody = redemptions.redeemStackedDiscounts(requestBody);
+            String responseBodyJson = JsonHelper.getObjectMapper().writeValueAsString(responseBody);
+            String snapshot = JsonHelper.readJsonFile(snapshotPath);
+            assertNotNull(responseBody);
+            JSONAssert.assertEquals(snapshot, responseBodyJson, false);
+        } catch (ApiException | IOException | JSONException | JsonSyntaxException e) {
+            fail();
+        }
+    }
+
+    @NotNull
+    private static RedemptionsRedeemRequestBody getRedemptionsRequestBody(int multipleVouchers) {
```

Comment:
`multipleVouchers` brzmi jak nazwa dla bool'a. A to jest jakiś count np. może `voucherCount` lub coś tego typu będzie lepsze.